models/sesemi.py

```diff
@@ -14,31 +14,48 @@
 # ========================================================================
 import torch
 import torch.nn as nn
-
-from .resnet import Resnet
-from .inception import Inception3
-from .densenet import Densenet
-from .efficientnet import EfficientNet
-
+from .timm import PyTorchImageModels
 import logging
 
 
 SUPPORTED_BACKBONES = (
-    'resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152',
-    'resnext50_32x4d', 'resnext101_32x8d', 'wide_resnet50_2', 'wide_resnet101_2',
-    'inception_v3', 'densenet121', 'densenet169', 'densenet201', 'densenet161',
+    # The following backbones strike a balance between accuracy and model size, with optional
+    # pretrained ImageNet weights. For a summary of their ImageNet performance, see
+    # <https://github.com/rwightman/pytorch-image-models/blob/master/results/results-imagenet.csv>.
 
-    # Standard EfficientNet models.
+    # Residual models and variants.
+    # Compared with the defaults, the "d" variants (e.g., resnet50d, resnest50d)
+    # replace the 7x7 conv in the input stem with three 3x3 convs.
+    # And in the downsampling block, a 2x2 avg_pool with stride 2 is added before conv,
+    # whose stride is changed to 1.
+    # Described in `Bag of Tricks <https://arxiv.org/abs/1812.01187>`.
```

Comment:
There looks to be a new set of ResNet models supported by timm based on the following paper:
[Revisiting ResNets: Improved Training and Scaling Strategies](https://arxiv.org/abs/2103.07579)

They are listed as:
resnetrs50, resnetrs101, resnetrs152, resnetrs200, resnetrs270, resnetrs350, resnetrs420

Would these be worthwhile including here?

Reply:
Up to resnetrs200, yes. The others are too big for their size with little accuracy gains.